### PR TITLE
Migrate endianconv.c unit tests to new test framework

### PR DESCRIPTION
**Pull request by karthyuom**
This PR migrates all tests related to endianconv into new test framework as part of the parent issue https://github.com/valkey-io/valkey/issues/428.

### CODE REVIEW

File: src/unit/test_endianconv.c [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    memrev16(buf);
+    printf("%s\n", buf);
+
+    snprintf(buf,sizeof(buf),"ciaoroma");
+    memrev32(buf);
+    printf("%s\n", buf);
+
+    snprintf(buf,sizeof(buf),"ciaoroma");
+    memrev64(buf);
+    printf("%s\n", buf);
```
Comment (madolson) [inline src/unit/test_endianconv.c]: Would you mind converting these to asserts. As of right now these don't actually validate anything.

Comment (karthyuom) [reply src/unit/test_endianconv.c]: @madolson right, added test assertions.

**Comment by karthyuom** [comment]
I didn’t remove the legacy endianconvTest from  _serverTest_ (see commit https://github.com/valkey-io/valkey/pull/458/commits/1c59ca7036977e06ad80de71cc501ea968071dc9 ) since it gives conflict which I couldn’t resolve anyway. I thought of doing it in a subsequent PR. I am not sure whethere there is any other alternative.

**Comment by karthyuom** [comment]
@madolson could you please review and merge the above, thanks.

File: src/unit/test_endianconv.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    snprintf(buf,sizeof(buf),"ciaoroma");
+    memrev32(buf);
+    TEST_ASSERT(!strcmp(buf, "oaicroma"));
+
+    snprintf(buf,sizeof(buf),"ciaoroma");
+    memrev64(buf);
+    TEST_ASSERT(!strcmp(buf, "amoroaic"));
+
+    return 0;
+}
```
Comment (madolson) [inline src/unit/test_endianconv.c]:
```suggestion
}

```

**Review by madolson — APPROVED** [review]

**Pull request merged**